ONSdigital/eq-questionnaire-launcher
# Ensure boolean parameters in quick launch are typed correctly

### What is the context of this PR?
When working on https://github.com/ONSdigital/eq-questionnaire-launcher/pull/64 I noticed that  when using quick launch, adding a parameter  to a URL which is intended to be a boolean would be a string in the token (incorrect type). In order to correct this, I had to add a method in to create a metadata object and add it to requiredSchemaMetadata in order for it to be casted to a boolean (which is done by the logic that is already in place).

(Note: When reviewing, the changes made is the refactoring of the solution I used to this problem in https://github.com/ONSdigital/eq-questionnaire-launcher/pull/64.)

### How to review
- Check the changes made are appropriate
- Run Runner in debug mode and add a breakpoint in line 110 in `session.py`. Then use quick launch and add a parameter to the URL such as `test=false` or `test=true` and see if the type in the token. (Note: The values of true and false are case insensitive so using `test=True` for example, will still result in a boolean in the token).
